common-types/src/main/scala/com/improving/app/common/errors/Validation.scala

```diff
@@ -1,85 +1,97 @@
 package com.improving.app.common.errors
 
-import com.improving.app.common.domain.{Address, CaPostalCodeImpl, Contact, MemberId, TenantId, UsPostalCodeImpl}
+import com.improving.app.common.domain.{Address, CaPostalCodeImpl, Contact, MemberId, OrganizationId, TenantId, UsPostalCodeImpl}
 
 object Validation {
 
-    type Validator[T] = T => Option[ValidationError]
+  type Validator[T] = T => Option[ValidationError]
 
-    def applyAllValidators[T](validators: Seq[Validator[T]]): Validator[T] =
-      (validatee: T) =>
-        validators.foldLeft[Option[ValidationError]](None)(
-          (maybeAlreadyError: Option[ValidationError], validator) =>
-            maybeAlreadyError.orElse(validator(validatee))
-        )
+  def applyAllValidators[T](validators: Seq[Validator[T]]): Validator[T] =
+    (validatee: T) =>
+      validators.foldLeft[Option[ValidationError]](None)(
+        (maybeAlreadyError: Option[ValidationError], validator) =>
+          maybeAlreadyError.orElse(validator(validatee))
+      )
 
-    def required[T]: (String, Validator[T]) => Validator[Option[T]] = (fieldName, validator) => {
-      opt =>
-        if (opt.isEmpty) {
-          Some(ValidationError(fieldName + " is missing."))
-        } else {
-          validator(opt.get)
-        }
+  def required[T]: (String, Validator[T]) => Validator[Option[T]] = (fieldName, validator) => {
+    opt =>
+      if (opt.isEmpty) {
+        Some(ValidationError(fieldName + " is missing."))
+      } else {
+        validator(opt.get)
+      }
+  }
+
+  def optional[T]: Validator[T] => Validator[Option[T]] = validator => opt => opt.flatMap(validator)
+
+  def skipEmpty(validator: Validation.Validator[String]): Validator[String] = str => {
+    if(str.isEmpty) {
+      None
+    } else {
+      validator(str)
     }
+  }
 
-    def optional[T]: Validator[T] => Validator[Option[T]] = validator => opt => opt.flatMap(validator)
+  val urlValidator: Validator[String] = url => {
+    None // TODO
```

Comment:
please create a ticket for this and remove the TODO

Reply:
I created IA-168 to flesh shared validation code

Reply:
sounds good